### ISSUE TITLE
Only listening on tcp6?

### ISSUE THREAD

**Issue by brainstorm** [sheepkiller/kafka-manager-docker#6]
Hello @sheepkiller,

I'm using the following `docker-compose` YAML which pulls your kafka-manager:

https://gist.github.com/chris-zen/19baa29a90625edef4db

Unfortunately it only seems to listen on tcp6 instead of plain old IPv4 tcp:

```
ubuntu@localhost:~$ netstat -tunl
(...)
tcp6       0      0 :::9000                 :::*                    LISTEN

ubuntu@localhost:~$ docker version
Client:
 Version:      1.9.1
 API version:  1.21
 Go version:   go1.4.2
 Git commit:   a34a1d5
 Built:        Fri Nov 20 13:16:54 UTC 2015
 OS/Arch:      linux/amd64

Server:
 Version:      1.9.1
 API version:  1.21
 Go version:   go1.4.2
 Git commit:   a34a1d5
 Built:        Fri Nov 20 13:16:54 UTC 2015
 OS/Arch:      linux/amd64

ubuntu@localhost:~$ uname -a
Linux localhost 3.19.0-21-generic #21+charmers1-Ubuntu SMP Tue Jun 16 10:41:47 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
```

Any ideas on why? Ping @chris-zen.

**Comment by sheepkiller** [comment]
Hi @brainstorm / @chris-zen,

Sorry for the late reply. I didn't test it on ubuntu. Have you tried setting "-Djava.net.preferIPv4Stack=true" as `args` ? 
thanks

**Comment by brainstorm** [comment]
Sorry, but which `args` are you referring to? Build time args for kafka-manager docker image? Couldn't find suitable ones for docker-compose, neither in the `docker-compose.yml` nor in its cmdline :-S

**Comment by sheepkiller** [comment]
Sorry I'm not familiar with docker-compose and I misread the documentation.
I forked your gist and add `command` parameters to force java to bind on the IPv4 address, inside the container. But  after investigations, it's pretty useless ; I  suspect `netstat` or `ss` to not expose containerized processes, what we're observing is the docker IPv6 proxy. I'm not - again - enough familiar with docker internals.

$ sudo ss -lpn | grep 9000
tcp    LISTEN     0      128      :::9000                 :::\*                   users:(("exe",pid=12770,fd=6))
$ ps auxwww | grep 12770
root     12770  0.0  0.1 395256 27092 ?        Sl   08:25   0:00 docker-proxy -proto tcp -host-ip 0.0.0.0 -host-port 9000 -container-ip 172.17.0.4 -container-port 9000

IPv4 is forwarded via iptables 
$ sudo iptables -L -n | grep 9000
ACCEPT     tcp  --  0.0.0.0/0            172.17.0.4           tcp dpt:9000

and for the sake of completeness, here what we have inside the container:
[root@a8615af66cab kafka-manager-1.3.0.4]# ss -lpn | grep 9000
tcp    LISTEN     0      50        _:9000                  *:_                   users:(("java",pid=1,fd=138))

thanks for reporting the issue, I learned something and I will modify my dockerfile to support arguments via an environment variable

EDIT: docker-compose gist: https://gist.github.com/sheepkiller/566de4698c9ed990095c

**Comment by brainstorm** [comment]
Thanks a ton! Will try your new docker image for `kafka-master` should you build one soon enough with that tweak! :+1:

**Comment by sheepkiller** [comment]
Hi,

Feel free to test unstable image/branch (which support kafka 0.9.0.x)
https://github.com/sheepkiller/kafka-manager-docker/tree/unstable
based on you're docker-compose file, here's an exemple : https://gist.github.com/sheepkiller/566de4698c9ed990095c

**Comment by chris-zen** [comment]
Definitively I will test the unstable image for Kafka 0.9.0.x, I've been waiting for it for some time.
Thank you so much.

**Comment by sheepkiller** [comment]
you're welcome. While I'm here, can I use your docker-compose file as a quick start ?
thanks.

**Comment by chris-zen** [comment]
Of course!
I recommend you to update the kafka image to the one from https://github.com/wurstmeister, he already merged the support for 0.9.0.1
You may also consider to adjust the kafka configuration to your needs.
Best
